### PR TITLE
Remove deprecated `opflow` module, `QuantumInstance` and related `utils`

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
This PR partially addresses https://github.com/Qiskit/qiskit/issues/10991 by removing `qiskit.opflow`, `qiskit.utils.quantum_instance.py`,  and related utils:

- `qiskit/utils/backend_utils.py`
- `qiskit/utils/measurement_error_mitigation.py`
- `qiskit/utils/mitigation/*`
- `qiskit/utils/run_circuits.py`

As well as removing their unit tests, updating the related sphinx references, and removing the API reference entry.

### Details and comments
This is PR # 3 of the *"Opflow & Algorithms deprecation series"*, it is on hold until https://github.com/Qiskit/qiskit/pull/11086 is merged.

### CODE REVIEW

**Review by Eric-Arellano — COMMENTED** [review]
FYI this will break some cross-references for the migration guides. That's fine because the migration guides are ported already to https://docs.quantum-computing.ibm.com/api/migration-guides and we're planning to switch the `qisk.it` short links this week to them. The IBM docs will update their links to use the `0.44` version of the API docs

**Comment by woodsp-ibm** [comment]
FYI: A PR is updating backend_utils for Aer 0.13 compatibility and is designated to be backported Qiskit/qiskit#11172 so I imagine we will want that to merge before the module gets deleted by this PR

File: qiskit/circuit/library/evolved_operator_ansatz.py [hunk 1]
```diff
@@ -147,7 +139,7 @@ def operators(self):
     def operators(self, operators=None) -> None:
         """Set the operators to be evolved.
 
-        operators (Optional[Union[OperatorBase, QuantumCircuit, list]): The operators to evolve.
+        operators (Optional[Union[QuantumCircuit, list]): The operators to evolve.
```
Comment (jakelishman) [inline qiskit/circuit/library/evolved_operator_ansatz.py]:
```suggestion
        operators (Optional[Union[QuantumCircuit, list]]): The operators to evolve.
```

File: qiskit/qpy/__init__.py [hunk 2]
```diff
@@ -921,7 +921,7 @@
 SPARSE_PAULI_OP_LIST_ELEM
 -------------------------
 
-This represents an instance of :class:`.PauliSumOp`.
+This represents an instance of :class:`.SparsePauliOp`.
```
Comment (jakelishman) [inline qiskit/qpy/__init__.py:924]: lol

File: test/python/circuit/library/test_evolution_gate.py [hunk 3]
```diff
@@ -247,9 +241,8 @@ def test_cnot_chain_options(self, option):
 
     @data(
         Pauli("XI"),
-        X ^ I,  # PauliOp
+        X ^ I,  # SparsePauliOp
```
Comment (jakelishman) [inline test/python/circuit/library/test_evolution_gate.py]:
```suggestion
```

we'd just be duplicating the explicit `SparsePauliOp` test that's the line below this one

File: test/python/result/test_sampled_expval.py [hunk 4]
```diff
@@ -82,17 +81,12 @@ def test_same(self):
...
-        with self.assertWarns(DeprecationWarning):
-            exp4 = sampled_expectation_value(counts, PauliSumOp(spo, coeff=2))
-        self.assertAlmostEqual(exp4, 2 * ans)
+        exp3 = sampled_expectation_value(counts, spo)
+        self.assertAlmostEqual(exp3, ans)
 
-        exp5 = sampled_expectation_value(counts, SparsePauliOp.from_list([[oper, 1]]))
-        self.assertAlmostEqual(exp5, ans)
+        exp4 = sampled_expectation_value(counts, SparsePauliOp.from_list([[oper, 1]]))
+        self.assertAlmostEqual(exp4, ans)
```
Comment (jakelishman) [inline test/python/result/test_sampled_expval.py]:
```suggestion
```

this is the same test as `exp3` above now, just spelled differently

**Review by jakelishman — COMMENTED** [review]
Thanks for doing all this, Elena.  It looks like we're on the verge of being ready to go, with overzealous type hints in Aer 0.13 being the only thing that prevents Neko from passing and this from merging.

With Qiskit/qiskit-aer#1996 now merged, we're just waiting on the 0.13.1 patch release of Aer, which at the time of writing, looks just about ready to go: Qiskit/qiskit-aer#1983.

File: test/python/result/test_sampled_expval.py (hunk 4 above)
Comment (ElePT) [reply test/python/result/test_sampled_expval.py]: oh, I thought I had removed that repetition, thx!

**Review by jakelishman — DISMISSED** [review]
Thanks for all this!  I think this PR is completely ready to merge from our side, we're just waiting for Aer 0.13.1 so Neko can pass.

**Comment by woodsp-ibm** [comment]
From above, since its still not happened

> FYI: A PR is updating backend_utils for Aer 0.13 compatibility and is designated to be backported https://github.com/Qiskit/qiskit/pull/11172 so I imagine we will want that to merge before the module gets deleted by this PR

**Comment by ElePT** [comment]
> From above, since its still not happened
> 
> > FYI: A PR is updating backend_utils for Aer 0.13 compatibility and is designated to be backported #11172 so I imagine we will want that to merge before the module gets deleted by this PR

Approved and added to the merge queue!

**Review by 1ucian0 — APPROVED** [review]

**Comment by jakelishman** [comment]
On hold until Aer 0.13.1 is released.

**Pull request merged**